In [4]:
import pandas as pd
import numpy as np

# Importing data

In [2]:
train_in_df = pd.read_csv('data/Train_Inpatientdata-1542865627584.csv')
train_out_df = pd.read_csv('data/Train_Outpatientdata-1542865627584.csv')
train_df = pd.read_csv('data/Train-1542865627584.csv')

In [3]:
print(f'Train_Inpatientdata shape: {train_in_df.shape}')
print(f'Train_Outpatientdata shape: {train_out_df.shape}')
print(f'Train shape: {train_df.shape}')

Train_Inpatientdata shape: (40474, 30)
Train_Outpatientdata shape: (517737, 27)
Train shape: (5410, 2)


In [13]:
# Common columns among all dfs
np.intersect1d(train_df.columns, np.intersect1d(train_in_df.columns, train_out_df.columns))


array(['Provider'], dtype=object)

In [14]:
# Common columns among train_in_df and train_out_df
np.intersect1d(train_in_df.columns, train_out_df.columns)

array(['AttendingPhysician', 'BeneID', 'ClaimEndDt', 'ClaimID',
       'ClaimStartDt', 'ClmAdmitDiagnosisCode', 'ClmDiagnosisCode_1',
       'ClmDiagnosisCode_10', 'ClmDiagnosisCode_2', 'ClmDiagnosisCode_3',
       'ClmDiagnosisCode_4', 'ClmDiagnosisCode_5', 'ClmDiagnosisCode_6',
       'ClmDiagnosisCode_7', 'ClmDiagnosisCode_8', 'ClmDiagnosisCode_9',
       'ClmProcedureCode_1', 'ClmProcedureCode_2', 'ClmProcedureCode_3',
       'ClmProcedureCode_4', 'ClmProcedureCode_5', 'ClmProcedureCode_6',
       'DeductibleAmtPaid', 'InscClaimAmtReimbursed',
       'OperatingPhysician', 'OtherPhysician', 'Provider'], dtype=object)

In [15]:
# Columns only in train_df
[l for l in train_df.columns if l not in train_in_df.columns and l not in train_out_df.columns]

['PotentialFraud']

In [16]:
# Columns only in train_in_df
[l for l in train_in_df.columns if l not in train_df.columns and l not in train_out_df.columns]

['AdmissionDt', 'DischargeDt', 'DiagnosisGroupCode']

In [17]:
# Columns only in train_out_df
[l for l in train_out_df.columns if l not in train_in_df.columns and l not in train_df.columns]

[]

traind_df is the table containing the target variable but it only has 5410 rows. 
Do we only have 5410 to work or are these Providers duplicated among train_in_df and train_out_df

In [27]:
print(f'Unique providers in train_in_df: {len(train_in_df.Provider.unique())}')
print(f'Unique providers in train_out_df: {len(train_out_df.Provider.unique())}')
print(f'Unique providers in train_df: {len(train_df.Provider.unique())}')



Unique providers in train_in_df: 2092
Unique providers in train_out_df: 5012
Unique providers in train_df: 5410


It looks that Providers are duplicated in theese data frames. It looks that we will use it only as a merge column. 
Does it make sense that a provider will allways be fraudulent??? in all 'transacations'?? 
I think that the idea is that if one provider has commited fraud he will be potentially fraudulent for ever and that is what we have to predict. So the goal of our model will be to determine potential fraud and not fraud directly. It still makes sense as the research on fraud time can be reduced a lot